src/features/references/editor/ReferenceDetailsCard.tsx

```diff
@@ -0,0 +1,261 @@
+import { createContext, Dispatch, ReactElement, SetStateAction, useCallback, useContext, useState } from 'react';
+
+import { CloseIcon, EditIcon } from '../../../components/icons';
+import { cx } from '../../../lib/cx';
+import { Author, ReferenceItem } from '../../../types/ReferenceItem';
+
+interface IEditingContext {
+  editing: boolean;
+  setEditing: Dispatch<SetStateAction<boolean>>;
+}
+
+const EditingContext = createContext<IEditingContext>({
+  editing: false,
+  setEditing: () => {
+    console.log('seting edtiting');
+  },
+});
+
+const returnFormatedAuthorsString = (authorsArray: Author[]): string => {
+  let authorsString = '';
+  authorsArray.forEach((a, index) => {
+    authorsString += (index > 0 ? ', ' : '') + a.fullName;
+  });
+  return authorsString;
+};
+
+const getContentString = (content: string | Author[] | undefined, id: string): string => {
+  if (typeof content == 'undefined') {
+    return '';
+  } else if (typeof content != 'string' && id === 'authors') {
+    return returnFormatedAuthorsString(content);
+  } else if (id === 'citationKey') {
+    return '[' + content.toString() + ']';
+  } else {
+    return content.toString();
+  }
+};
+
+const DataTextInput = ({
+  content,
+  id,
+  onChangeHandler,
+}: {
+  content: string;
+  id: string;
+  onChangeHandler: CallableFunction;
```

Comment:
Please use the explicit function type `(id: string, value: string) => void` instead of `CallableFunction`.
Also nit comment: the convention is usually to just call it `onChange` and in this case, maybe `onSave` or `onBlur` would be better since this is not called in the change handler of the input